lib/engine/step/g_1822/buy_train.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require_relative '../buy_train'
+
+module Engine
+  module Step
+    module G1822
+      class BuyTrain < BuyTrain
+        def actions(entity)
+          return [] if entity != current_entity
+
+          return %w[sell_shares buy_train] if president_may_contribute?(entity)
+          return %w[buy_train pass] if can_buy_train?(entity)
```

Comment:
Is this for minors? Does 1867 do the same?

Reply:
Was a long time since i played 1867 so i dont remember. The code is not done, i just took the base function and removed the
`return [] if entity.minor?`
so the game doesnt crash.

Reply:
The good news is you may be able to use a lot of Jen's 1867 code for minors. I haven't looked into how she implemented it, so don't know for sure.

Reply:
1867 minors aren't actually minors but real corporations with a type set to :minor, :major's are major's

Reply:
1867 minors needed share price and not going down this approach was going to be quite painful.

Reply:
@magnusfalt, I recommend going the route @jenf did. It sounds like it'll make for an easier implementation. The Minors class is intended for non-share minors like in 1846.